src/MethodExecutor.cs

```diff
@@ -15,94 +15,42 @@
 // You should have received a copy of the GNU General Public License
 // along with Gauge-CSharp.  If not, see <http://www.gnu.org/licenses/>.
 
-using System.Collections.Generic;
-using System.Diagnostics;
-using Gauge.CSharp.Core;
-using Gauge.CSharp.Lib;
-using Gauge.Dotnet.Models;
-using Gauge.Dotnet.Strategy;
-using Gauge.Messages;
-using Google.Protobuf;
+using System;
+using System.Reflection;
+using Gauge.Dotnet.Wrappers;
 using NLog;
 
 namespace Gauge.Dotnet
 {
-    public class MethodExecutor : IMethodExecutor
+    public class MethodExecutor
```

Comment:
There are no concrete instances of MethodExecutor, and is only inherited by `StepExecutor` and `HookExecutor`. Consider making this `abstract`.

After this and other points, see if there is a need to have `IStepExecutor` and `IHookExecutor` interface. Even if they are consolidated to `IExecutor`, you may see if `MethodExecutor` can inline the `Execute` as an `abstract` method.